src/main/java/io/supertokens/webserver/api/multitenancy/BaseRemove.java

```diff
@@ -46,7 +46,7 @@ protected void handle(TenantIdentifier sourceTenantIdentifier, TenantIdentifier
         boolean didExist = false;
 
         try {
-            Multitenancy.checkSourceAndTargetTenantForCreateOrUpdate(sourceTenantIdentifier, targetTenantIdentifier);
+            Multitenancy.checkPermissionsForCreateUpdateOrDelete(main, sourceTenantIdentifier, targetTenantIdentifier);
```

Comment:
recheck deletion permissions.. what happens if i try to delete the tenant that i am querying from? Is that allowed or not?

Reply:
The permission checks should be the same as the list APIs

Reply:
You can't remove the public tenantId from the tenant remove API.